pkg/imagestream/imagestreamhasblob.go

```diff
@@ -37,6 +38,27 @@ func (is *imageStream) HasBlob(ctx context.Context, dgst digest.Digest, requireM
 		return found
 	}
 
+	if layers, err := is.imageStreamGetter.layers(); err == nil {
+		if _, ok := layers.Blobs[dgst.String()]; !ok {
+			return logFound(nil)
+		}
+		copied := make([]imageapiv1.ImageLayer, 0, len(layers.Blobs))
+		for name, blob := range layers.Blobs {
+			copied = append(copied, imageapiv1.ImageLayer{Name: name, LayerSize: blob.LayerSize, MediaType: blob.MediaType})
+		}
+		var firstImage string
+		for name := range layers.Layers {
+			firstImage = name
```

Comment:
we don't actually need to find the specific image that contains the blob?  any image in the imagestream will do?

Reply:
correct

Reply:
If we have the list of blobs, we don't need the image object anymore, we should just return that list.